src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -940,21 +1027,12 @@ object LoadVCF {
           }
         }
         rvb.endArray()
-      }(lines, rowType, rg, contigRecoding, arrayElementsRequired, skipInvalidLoci),
-      justVariants)
+      }(lines, rvRowType, rg, contigRecoding, arrayElementsRequired, skipInvalidLoci))
 
-    new MatrixTable(hc,
-      matrixType,
+    MatrixValue(matrixType,
       BroadcastRow(Row.empty, matrixType.globalType, sc),
-      BroadcastIndexedSeq(sampleIds.map(x => Annotation(x)), TArray(matrixType.globalType), sc),
-      rdd)
+      BroadcastIndexedSeq(localSampleIDs.map(x => Annotation(x)), TArray(matrixType.globalType), sc),
```

Comment:
prefer `Annotation(_)`

Reply:
I think this should be `matrixType.colType`, may explain test failure

Reply:
this was clearly busted before too. I'm not sure how it was passing.